libraries/AP_Notify/ToneAlarm.cpp

```diff
@@ -137,11 +134,14 @@ void AP_ToneAlarm::_timer_task()
 void AP_ToneAlarm::play_string(const char *str)
 {
     WITH_SEMAPHORE(_sem);
-
     _mml_player.stop();
-    strncpy(_tone_buf, str, AP_NOTIFY_TONEALARM_TONE_BUF_SIZE);
-    _tone_buf[AP_NOTIFY_TONEALARM_TONE_BUF_SIZE-1] = 0;
-    _mml_player.play(_tone_buf);
+    
+    if (pNotify->buzzer_enabled())
```

Comment:
Same here (calling `stop_cont_tone` when disabled) - doing this change means you need to check on `init` whether to play startup tone or not.